tls/s2n_config.h

```diff
@@ -121,6 +121,8 @@ struct s2n_config {
     uint32_t server_max_early_data_size;
 
     s2n_psk_mode psk_mode;
+
+    s2n_async_pkey_validation_mode async_pkey_validation_mode;
 };
```

Comment:
Why only validate the signature when signing asynchronously? Could the same problem happen with synchronous signatures?

Reply:
It should not as we validate that private key matches public key in [here](https://github.com/aws/s2n-tls/blob/main/crypto/s2n_certificate.c#L340). But for async pkey, the application may use a different private key to produce a signature by mistake.